test/multi-ns/connection_test.go

```diff
@@ -0,0 +1,155 @@
+		"Connect to one NSE": {
+			[]string{"foo"},
+			map[string]string{"ns.networkservicemesh.io": "foo?app=vl3-nse-foo"},
+		},
+		"Connect to two NSEs": {
+			[]string{"foo", "bar"},
+			map[string]string{"ns.networkservicemesh.io": "foo?app=vl3-nse-foo,bar?app=vl3-nse-bar"},
+		},
+	} {
+		t.Logf("Running test case: %s", testName)
+		var deployments []appsv1.Deployment
+		var ip_addr string
+		flag.StringVar(&ip_addr, "REMOTE_IP", "127.0.0.1", "Set ENV to REMOTE_IP")
+		flag.Parse()
+		//Install network service
+		for _, name := range c.networkServices {
+			fmt.Printf("Installing Network Service %s\n", name)
+			os.Setenv("REMOTE_IP", ip_addr)
+			os.Setenv("SERVICENAME", name)
+			cmd := exec.Command("../../scripts/vl3/vl3_interdomain.sh")
+			err := cmd.Run()
+			if err != nil {
+				fmt.Println(err.Error())
+				return
+			}
+		}
+		endpoints := listDeployments("wcm-system", clientset)
+		deployments = append(deployments, endpoints...)
+
+		//Create client pod with annotation
+		clientDep.ObjectMeta.Annotations = c.annotations
+		fmt.Println("Create client deployment")
+		deploymentsClient := clientset.AppsV1().Deployments(corev1.NamespaceDefault)
+		result, err := deploymentsClient.Create(context.TODO(), clientDep, metav1.CreateOptions{})
+		if err != nil {
+			fmt.Println(err.Error())
+			return
+		}
+		deployments = append(deployments, *result)
+
+		var ready bool
+		for i := 0; i < 12; i++ {
+			fmt.Println("Waiting for client pod to start...")
+			time.Sleep(time.Duration(5) * time.Second)
+			ready = checkAvailability(result, clientset)
+			if ready {
+				break
+			}
+		}
+		if !ready {
+			t.Error("Client pod NOT connected to networkservices!")
+		} else {
+			t.Log(("Success! Client pod connected to networkservices!"))
+		}
+
+		//Cleanup deployments
+		for _, d := range deployments {
+			removeDeployment(d.ObjectMeta.Namespace, d.ObjectMeta.Name, clientset)
+		}
+	}
+}
+
+func removeDeployment(namespace, deploymentName string, clientset *kubernetes.Clientset) {
+	deploymentsClient := clientset.AppsV1().Deployments(namespace)
+	if err := deploymentsClient.Delete(context.TODO(), deploymentName, metav1.DeleteOptions{}); err != nil {
+		fmt.Println(err.Error())
+	}
+	fmt.Printf("Deleted deployment %s\n", deploymentName)
+}
+
+func checkAvailability(deployment *appsv1.Deployment, clientset *kubernetes.Clientset) bool {
+	namespace := deployment.ObjectMeta.Namespace
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
+	for _, pod := range pods.Items {
+		if pod.Status.Phase != corev1.PodRunning {
+			return false
+		}
+	}
+	return true
+}
+
+func listDeployments(namespace string, clientset *kubernetes.Clientset) []appsv1.Deployment {
+	deploymentsClient := clientset.AppsV1().Deployments(namespace)
+	list, err := deploymentsClient.List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+	return list.Items
+}
```

Comment:
1. Please make sure you provide Valuable messages.
For example at lines 125-127. You are printing the error but that doesn't mean it would help me discover where the problem is.

2. Line 125-127 if you decide that you want to print the error and still continue executing the program, please make sure that line 128 doesn't execute by adding a return after the print at line 126. That way we would not end up with logs that tell that the deployment was deleted when in fact that didn't happen.  

3. The best approach for all of those functions would be to add an error in the returned type and catch it in the main test. Also make sure that you add a message that would be helpful to understand where the error happend and why.

4. In the ```checkAvailability``` function when you get an error that would mean you would not get any pods in the ```pods``` variable so you could just return the proper boolean value there.

5. In the ``listDeployments`` function is the same as ``4.`` if you get an error please make sure you handle it or propagate it further and catch it in the for block.